clang/lib/Sema/SemaExpr.cpp

```diff
@@ -2347,6 +2347,27 @@ Sema::BuildDeclRefExpr(ValueDecl *D, QualType Ty, ExprValueKind VK,
   return E;
 }
 
+// Check whether a similar function-like macro exists and suggest it
```

Comment:
```suggestion
// Diagnose when a macro cannot be expanded because it's a function-like macro
// being used as an object-like macro. Returns true if a diagnostic is emitted.
```